PeerioTechnologies/peerio-mobile
# Feat tap inline file action

### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/4406/mobile-open-download-if-needed-file-when-tapping-on-filename-or-icon
https://app.clubhouse.io/peerio/story/6759/mobile-file-options-missing

While developing this found a bug https://app.clubhouse.io/peerio/story/5462/mobile-android-upload-image-to-chat-and-try-opening-the-image-from-inline-options-does-nothing

#### EDIT:
This task started as "tap on name or icon to open or download" and turned into:

1- Tap on file name or icon to open/download

2- Tap on image preview to open/download (including large images, you can tap on the text)

3- Open files automatically after download is complete (only implemented in chat)

4- Since some file options were missing in files list and chats, it was decided that all action sheets related to files will be refactored to the custom action sheet, all changed action sheets are all listed here:
a- Files List
b- Inline Files (and images)
c- Recent Files list
d- File view (click on a file in files list)

I worked closely with Saumya while changing the action sheets to make sure that each of them has the right actions for this release. The only difference is that a file cannot be deleted in File View (at least for now).

One change is to be made when Internal Viewer is implemented; the "Open" action should not be listed in some of the action sheets.

Another UX change that was discussed was whether non-owners should be able to delete a file in chat. It was decided that deleting a file in chat by a non-owner might cause the user to think that they deleted the file for everyone. I've kept that option for now, please let me know if that needs to be changed. 

Since the all 4 action sheets listed above use the same logic, there is no need to test every action sheet's button. The main thing to be tested for the action sheets is that the buttons shown are logical from the user's point of view.

### Testing instructions  
4 Scenarios to test:

1- Send image from gallery in chat. Images smaller than 3 MB should be downloaded automatically, so pressing on the name/icon will do nothing. After the image is downloaded, pressing on the button opens the image. If image is not downloaded automatically, clicking on the name/icon will start a download. (the bug mentioned above means that this scenario doesn't behave correctly on some Android versions)

2- Send file from gallery to chat. File that has been shared in chat needs to be downloaded before viewing. Clicking on name/icon after sharing will begin the download, and after the download is finished, file will automatically open (there is a bug currently where files open using the wrong viewer CH#6607). Clicking during download has no effect. Clicking after it has been downloaded opens file (on Android this depends on what program is your default for opening files).

3- Receive image in chat. Same as 1, except it doesn't bug on Android.
4- Receive file in chat. Same as 2.

Note: Since images mostly auto download (except the ones that are too big), I did not add an automatic open after download for images. Please let me know if there is a case that I missed where auto-open after download is needed.

#### EDIT:
Please refer to the "Relevant info and issue/PR links".

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
